ghcide/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -1385,35 +1406,89 @@ newImportToEdit (unNewImport -> imp) ps fileContents
 -- * If the file has neither existing imports nor a module declaration,
 -- the import will be inserted at line zero if there are no pragmas,
 -- * otherwise inserted one line after the last file-header pragma
-newImportInsertRange :: ParsedSource -> T.Text -> Maybe (Range, Int)
-newImportInsertRange (L _ HsModule {..}) fileContents
+newImportInsertRange :: Annotated ParsedSource -> T.Text -> Maybe (Range, Int)
+newImportInsertRange ps fileContents
   |  Just ((l, c), col) <- case hsmodImports of
-      [] -> findPositionNoImports (fmap reLoc hsmodName) (fmap reLoc hsmodExports) fileContents
-      _  -> findPositionFromImportsOrModuleDecl (map reLoc hsmodImports) last True
+      [] -> (\line -> ((line, 0), 0)) <$> findPositionNoImports ps fileContents
```

Comment:
What is this doing? why is it okay?

Reply:
Previously, it returns `((line, column), indent)` (the position for new import). But `column` and `indent` are always zero, so I simplified it a bit, to just return the line number.